sim.py

```diff
@@ -51,26 +51,39 @@ def write_data(file, data):
         writer.writerows(data)
 
 
-def test_characters(characters, start: int, end: int, extra_args: Dict[str, bool]):
+class Sim:
+    def __init__(self, name: str, constructor, **kwargs):
+        self.name = name
+        self.constructor = constructor
+        self.args = kwargs
+
+
+def test_characters(characters: List[Sim], start: int, end: int, extra_args: Dict[str, bool]):
     data = [["Level", "Character", "DPR"]]
     for level in range(start, end + 1):
-        for [name, Creator] in characters:
-            data.append([level, name, test_dpr(Creator(level, **extra_args), level)])
+        for character in characters:
+            args = dict(extra_args)
+            args.update(character.args)
+            data.append([level, character.name, test_dpr(character.constructor(level, **args), level)])
     return data
 
 
+def sim(name: str, constructor, **kwargs):
+    return Sim(name, constructor, **kwargs)
+
+
 CHARACTER_MAPPING = {
-    "monk": ["Monk", Monk],
-    "champion": ["Champion Fighter", ChampionFighter],
-    "battlemaster": ["Battlemaster Fighter", ChampionFighter],
-    "barbarian": ["Barbarian", Barbarian],
-    "paladin": ["Paladin", Paladin],
-    "gloomstalker": ["Gloomstalker Ranger", GloomstalkerRanger],
-    "beastmaster": ["Beastmaster Ranger", BeastMasterRanger],
-    "rogue": ["Rogue", AssassinRogue],
-    "wizard": ["Wizard", Wizard],
-    "cleric": ["Cleric", Cleric],
-    "au": ["Assault Unit 2 1", AssaultUnit],
+    "monk": sim("Monk", Monk),
+    "champion": sim("Champion Fighter", ChampionFighter),
+    "battlemaster": sim("Battlemaster Fighter", ChampionFighter),
+    "barbarian": sim("Barbarian", Barbarian),
+    "paladin": sim("Paladin", Paladin),
+    "gloomstalker": sim("Gloomstalker Ranger", GloomstalkerRanger),
+    "beastmaster": sim("Beastmaster Ranger", BeastMasterRanger),
+    "assassin": sim("Assassin (BB)", AssassinRogue),
```

Comment:
We need the `rogue` config for the default `all` config below. Perhaps create a new one called `assassin`?